cmd/embedded-cluster/install.go

```diff
@@ -266,7 +266,7 @@ func getLicenseFromFilepath(licenseFile string) (*kotsv1beta1.License, error) {
 		// if the app is different, we will not be able to provide the correct vendor supplied charts and k0s overrides
 		return nil, fmt.Errorf("license app %s does not match binary app %s, please provide the correct license", license.Spec.AppSlug, rel.AppSlug)
 	}
-	if rel.ChannelID != license.Spec.ChannelID {
+	if !channelIDExists(license.Spec.Channels, rel.ChannelID) {
```

Comment:
should we still handle pre-'channels' licenses? IIRC that field didn't always exist

Reply:
(plausibly we shouldn't as licenses should come with the binary download, but...)

Reply:
Well, our CI uses a license file like that 😄 

Make sure to update the error message below this too, and the unit testing

Reply:
@laverya heh yea sorry this was a WIP (still a draft)

Reply:
oh no this was just me going "oh hey PR from a username I don't recognize", no worries at all

Reply:
ah cool, my work flow is little commit spammy 😁 but i appreciate the early looks! @laverya think this is ready to go if ya wanna take another peek.